swap/swap_test.go

```diff
@@ -184,267 +184,197 @@ func testBalances(t *testing.T, swap *Swap, expectedBalances map[enode.ID]int64)
-	}
+// generates a cheque and adds it as the last received cheque for the given peer, fails if there are errors
+func setNewReceivedCheque(t *testing.T, p *Peer) *Cheque {
+	t.Helper()
+	return setNewCheque(t, p.setLastReceivedCheque)
+}
 
-	// check previous cheque is still correct
-	receivedCheque, err = swap.ReceivedCheque(testPeer.ID())
+func setNewCheque(t *testing.T, setChequeFunction func(*Cheque) error) *Cheque {
+	t.Helper()
+	generatedCheque := newRandomTestCheque()
+	err := setChequeFunction(generatedCheque)
 	if err != nil {
 		t.Fatal(err)
 	}
-	if receivedCheque != generatedCheque {
-		t.Fatalf("Expected received cheque to be %v, but is %v", generatedCheque, receivedCheque)
-	}
+	return generatedCheque
+}
 
-	// test received cheque for invalid peer
-	randomID := adapters.RandomNodeConfig().ID
-	_, err = swap.ReceivedCheque(randomID)
-	if err == nil {
-		t.Fatal("Expected call to fail, but it didn't!")
-	}
-	if err != state.ErrNotFound {
-		t.Fatalf("Expected test to fail with %s, but is %s", "ErrorNotFound", err.Error())
-	}
+// generates a cheque and saves it as the last sent cheque for a peer in the given swap struct, fails if there are errors
+func saveNewSentCheque(t *testing.T, s *Swap, id enode.ID) *Cheque {
+	t.Helper()
+	return saveNewCheque(t, id, s.saveLastSentCheque)
+}
 
-	// test received cheque for disconnected node
-	testPeer3 := newDummyPeer().Peer
-	generatedCheque3 := newRandomTestCheque()
-	err = swap.saveLastReceivedCheque(testPeer3.ID(), generatedCheque3)
+// generates a cheque and saves it as the last received cheque for a peer in the given swap struct, fails if there are errors
+func saveNewReceivedCheque(t *testing.T, s *Swap, id enode.ID) *Cheque {
+	t.Helper()
+	return saveNewCheque(t, id, s.saveLastReceivedCheque)
+}
+
+func saveNewCheque(t *testing.T, id enode.ID, saveChequeFunction func(enode.ID, *Cheque) error) *Cheque {
+	t.Helper()
+	generatedCheque := newRandomTestCheque()
+	err := saveChequeFunction(id, generatedCheque)
 	if err != nil {
 		t.Fatal(err)
 	}
-	receivedCheque3, err := swap.ReceivedCheque(testPeer3.ID())
+	return generatedCheque
+}
+
+// tests that a nested map of peerID:{typeOfCheque:cheque} matches the result of the Cheques function
+func testChequesByPeerAndType(t *testing.T, s *Swap, expectedCheques map[enode.ID]map[string]*Cheque) {
+	t.Helper()
+	cheques, err := s.Cheques()
 	if err != nil {
 		t.Fatal(err)
 	}
-	if !reflect.DeepEqual(receivedCheque3, generatedCheque3) {
-		t.Fatalf("Expected received cheque to be %v, but is %v", generatedCheque3, receivedCheque3)
+	if !reflect.DeepEqual(expectedCheques, cheques) {
+		t.Fatalf("Expected cheques to be %v, but are %v", expectedCheques, cheques)
 	}
 }
 
-// Test getting cheques for all known peers
-func TestAllCheques(t *testing.T) {
+// TestPeerCheques verifies that sent and received cheques data for a given peer is correct
+func TestPeerCheques(t *testing.T) {
+	testBackend := newTestBackend()
```

Comment:
this test should also be migrated to be a table-driven test

Reply:
awaiting reply to [previous response](https://github.com/ethersphere/swarm/pull/1863#discussion_r338799656)

Reply:
done